astropy/nddata/tests/test_ccddata.py

```diff
@@ -1081,3 +1082,48 @@ def test_read_write_tilde_paths(home_is_tmpdir):
     # Ensure the unexpanded path doesn't exist (e.g. no directory whose name is
     # a literal ~ was created)
     assert not os.path.exists(filename)
+
+
+def test_ccddata_with_psf():
+    psf = _random_psf.copy()
+    ccd = CCDData(_random_array.copy(), unit=u.adu, psf=psf)
+    assert (ccd.psf == psf).all()
+
+    # cannot pass in non-ndarray
+    with pytest.raises(TypeError, match="The psf must be a numpy array."):
+        CCDData(_random_array.copy(), unit=u.adu, psf="something")
+
+
+def test_psf_setter():
+    psf = _random_psf.copy()
+    ccd = CCDData(_random_array.copy(), unit=u.adu)
+    ccd.psf = psf
+    assert (ccd.psf == psf).all()
+
+    # cannot set with non-ndarray
+    with pytest.raises(TypeError, match="The psf must be a numpy array."):
+        ccd.psf = _random_array * u.pix
```

Comment:
```suggestion
        ccd.psf = 5
```

Reply:
But... do you really *want* to support Quantity? How will all the different units interact with this functionality?

Reply:
I think it is fine -- the PSF is just carried along sort of like the metadata. I don't see a reason to forbid `Quantity` given that. The uncertainties we need to pay attention because we do arithmetic with them but I think we are ok here with quantities without needing to force consistency of some sort between the units of the `CCDData` object and the unit (if any) of the `psf`.

Reply:
I am just worried that you will give wrong answer without realizing it if someone pass in Quantity with unit that somehow doesn't trip any exception. But if you are sure that is not a problem, then please change the test as you suggested. Thanks!

Reply:
Right now it isn't clear to me what unit to expect if a `Quantity` is passed in -- right now if any arithmetic is attempted with a `psf` attached we emit a warning.

Hopefully we get some user feedback on this after release.